app/notification_email.go

```diff
@@ -283,6 +287,36 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+	var publicChannels *model.ChannelList
+	if team, err := a.GetTeamByName(teamName); err != nil {
+		mlog.Error(fmt.Sprintf("Encountered error while looking up team by name for %s", teamName))
+		return postMessage
+	} else {
+		if publicChannels, err = a.GetPublicChannelsForTeam(team.Id, 0, 100); err != nil {
+			mlog.Error(fmt.Sprintf("Encountered error while looking up Public Channels for Team: %s", teamName))
+			return postMessage
+		}
+	}
+
+	r := regexp.MustCompile(`\~[\w-]+\b`)
```

Comment:
Instead of custom regex to find channel mentions, let's use the existing [ChannelMentions](https://github.com/mattermost/mattermost-server/blob/8cdbc734b98d2f521820cbd155259917b979f7f5/model/channel_mentions.go#L13-L28) method. 

This will return a list of names that we can then use with [GetChannelsByNames](https://github.com/mattermost/mattermost-server/blob/8cdbc734b98d2f521820cbd155259917b979f7f5/app/channel.go#L1190-L1201), allowing us to handle an arbitrary number of channels on the server.

Finally, we can replace all matches with the desired link.

Reply:
Yeah, I think this makes sense. Plus, it saves us a database call if there's no channel links in the post.

Reply:
I didn't realize they had the ChannelMentions method. Sorry about that. But one question about mentions, do they use DisplayName of the channel, Name of the channel or both in the post?

Reply:
It just uses the Name of the channel, but that's expected. It's like how at-mentions for users only look for the Username